File: src/test/java/com/letsdev/employeedevelopmentcards/integration/api/OpenapiControllerTest.java
package com.letsdev.employeedevelopmentcards.integration.api;

import com.letsdev.employeedevelopmentcards.integration.AbstractIntegrationTest;
import org.junit.Test;

import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;

public class OpenapiControllerTest extends AbstractIntegrationTest {

    @Test
    public void testOpenapiUiRedirect() throws Exception {
        mockMvc.perform(get("/")).andExpect(status().is3xxRedirection());
    }

    @Test
    public void testOpenapiYaml() throws Exception {
        mockMvc.perform(get("/openapi.yaml")).andExpect(status().isOk());
        mockMvc.perform(get("/openapi.yml")).andExpect(status().isOk());
    }

    @Test
    public void testOpenapiJson() throws Exception {
        mockMvc.perform(get("/openapi.json")).andExpect(status().isOk());
    }

}

File: src/test/java/com/letsdev/employeedevelopmentcards/integration/api/OpenapiControllerTest.kt
package com.letsdev.employeedevelopmentcards.integration.api

import com.letsdev.employeedevelopmentcards.integration.AbstractIntegrationTest
import org.junit.Test

import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get
import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status

class OpenapiControllerTest : AbstractIntegrationTest() {

    @Test
    fun testOpenapiUiRedirect() {
        mockMvc.perform(get("/")).andExpect(status().is3xxRedirection)
    }

    @Test
    fun testOpenapiYaml() {
        mockMvc.perform(get("/openapi.yaml")).andExpect(status().isOk)
        mockMvc.perform(get("/openapi.yml")).andExpect(status().isOk)
    }

    @Test
    fun testOpenapiJson() {
        mockMvc.perform(get("/openapi.json")).andExpect(status().isOk)
    }

}
